frontend/src/components/Container.js

```diff
@@ -19,207 +19,210 @@ import Header from './header/Header';
-          (noMargin ? 'dashboard' : 'container-fluid')
-        }
-      >
-        {modal.visible && (
-          <Modal
-            {...modal}
-            windowType={modal.type}
-            dataId={modal.dataId ? modal.dataId : dataId}
-            modalTitle={modal.title}
-            modalViewId={modal.viewId}
-            parentType={windowType}
-            parentDataId={dataId}
-            query={query}
-            viewId={query && query.viewId}
-            rawModalVisible={rawModal.visible}
-            indicator={indicator}
-            modalViewDocumentIds={modal.viewDocumentIds}
-            closeCallback={closeModalCallback}
-            modalSaveStatus={
-              modal.saveStatus && modal.saveStatus.saved !== undefined
-                ? modal.saveStatus.saved
-                : true
-            }
-            isDocumentNotSaved={
-              modal.saveStatus &&
-              !modal.saveStatus.saved &&
-              (modal.validStatus && !modal.validStatus.initialValue)
-            }
+    return (
+      <div>
+        {!hideHeader && (
+          // Forcing refresh component
+          <Header
+            docStatus={docActionElem}
+            windowId={windowType}
+            {...{
+              entity,
+              docStatusData,
+              docNoData,
+              docSummaryData,
+              handleDeletedStatus,
+              isDocumentNotSaved,
+              showIndicator,
+              viewId,
+              siteName,
+              showSidelist,
+              attachments,
+              actions,
+              references,
+              breadcrumb,
+              dataId,
+              dropzoneFocused,
+              notfound,
+              docId,
+              editmode,
+              handleEditModeToggle,
+              activeTab,
+            }}
           />
         )}
 
-        {rawModal.visible && (
-          <RawModal
-            modalTitle={rawModal.title}
-            modalDescription={rawModal.description}
-            allowedCloseActions={rawModal.allowedCloseActions}
-            windowType={rawModal.windowId}
-            viewId={rawModal.viewId}
-            masterDocumentList={masterDocumentList}
-          >
-            <div className="document-lists-wrapper">
-              <DocumentList
-                type="grid"
-                windowType={rawModal.windowId}
-                defaultViewId={rawModal.viewId}
-                viewProfileId={rawModal.profileId}
-                setModalTitle={setRawModalTitle}
-                setModalDescription={setRawModalDescription}
-                fetchQuickActionsOnInit={
-                  !(
+        {connectionError && <ErrorScreen />}
+
+        <div
+          className={
+            'header-sticky-distance js-unselect ' +
+            (noMargin ? 'dashboard' : 'container-fluid')
+          }
+        >
+          {modal.visible && (
+            <Modal
+              {...modal}
```

Comment:
IMHO it's not a good approach to pass the whole `modal` (i don't know what's inside).
Why not explicitly sending the properties which are needed?

Reply:
Yeah the thing is I'm not really refactoring this part, but the whole content of the file was formatted differently after switching to `PureComponent` class. Sure, it's probably not the best approach but it's a rabbit hole I don't want to go down at this moment. Cause I'll end up rewriting way more stuff than I should.

Reply:
I agree with Kuba on this one. We need to deliver this on schedule(today).

Reply:
roger

Reply:
roger